### PR TITLE
feat: Add madminer recipe

### PR DESCRIPTION
**Pull request by matthewfeickert**
* c.f. https://github.com/madminer-tool/madminer
* Recipe generated with grayskull via 'grayskull pypi madminer' and then edited to add additional information.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [N/A] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
I am willing to be a maintainer.

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: recipes/madminer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/madminer-{{ version }}.tar.gz
+  sha256: 17eedd09da79db371530336904f248e8cde00032a53e50bbe36cbbb92f1c0939
+
+build:
+  # * PyTorch does not support Windows
+  # * uproot3-methods does not support Python 3.12+
+  # * Non noarch packages should have python requirement without any version constraints
+  # so use or statements in skip
+  skip: true  # [win or py<38 or py>=312]
```
Comment (matthewfeickert) [inline recipes/madminer/meta.yaml:17]: Can't use `noarch` as there are no distributions of `pytorch` for Windows, which means that can't place version restrictions on `python` in the requirements, so need to use the `skip` to restrict Python to 3.8+ but not allow Python 3.12.

**Comment by matthewfeickert** [comment]
@conda-forge/help-python, ready for review!

File: recipes/madminer/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipes/madminer/meta.yaml:17]: It is OK for noarch package to fails on Windows with a missing dependency. We prefer noarch b/c it is faster to build and the package will be ready when the missing dependency is available on Windows.

Comment (matthewfeickert) [reply recipes/madminer/meta.yaml:17]: Ah okay I'll revert this and then tag you when I have this ready. :+1:

File: recipes/madminer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+requirements:
+  # Python 3.12+ is not allowed as uproot3_methods fails with
+  # AttributeError: 'FileFinder' object has no attribute 'find_module'.
+  # uproot3 and uproot3-methods are deprecated, so these should be removed
+  # in the future.
+  host:
+    - python >=3.8, <3.12
+    - pip
+  run:
+    - python >=3.8, <3.12
```
Comment (ocefpaf) [inline recipes/madminer/meta.yaml:29]: I've seen weird breakages b/c of spaces like this. In theory, it should not matter but better safe than sorry.

```suggestion
    - python >=3.8,<3.12
    - pip
  run:
    - python >=3.8,<3.12
```

**Comment by matthewfeickert** [comment]
Thanks for the review and help @ocefpaf!

**Comment by matthewfeickert** [comment]
Note to self: This PR generated https://github.com/conda-forge/madminer-feedstock

**Pull request merged**